contracts/MarginPool.sol

```diff
@@ -130,4 +159,39 @@ contract MarginPool {
             require(transferToUser(_asset[i], _user[i], _amount[i]), "MarginPool: Transfer to user failed");
         }
     }
+
+    /**
+     * @notice function to collect excess balance
+     * @dev can only be called by farmer address
+     * @param _asset asset address
+     * @param _receiver receiver address
+     * @param _amount amount to harvest
+     */
+    function farm(
+        address _asset,
+        address _receiver,
+        uint256 _amount
+    ) external onlyFarmer {
+        require(_receiver != address(0), "MarginPool: invalid receiver address");
+
+        uint256 externalBalance = ERC20Interface(_asset).balanceOf(address(this));
+        uint256 storedBalance = getStoredBalance(_asset);
+
+        require(_amount <= externalBalance.sub(storedBalance), "MarginPool: amount exceed limit");
+
+        ERC20Interface(_asset).transfer(_receiver, _amount);
+
+        emit AssetFarmed(_asset, _receiver, _amount);
+    }
+
+    /**
+     * @notice function to set farmer address
+     * @dev can only be called by MarginPool owner
+     * @param _farmer farmer address
+     */
+    function setFarmer(address _farmer) external onlyOwner {
+        emit FarmerUpdated(farmer, _farmer);
+
+        farmer = _farmer;
```

Comment:
```suggestion
        farmer = _farmer;

        emit FarmerUpdated(farmer, _farmer);

```

Reply:
can we just simplified it to `FarmerUpdated(farmer)` ?

Reply:
its nice to know the old and new tbh